src/HTTPRequest.cpp

```diff
@@ -11,20 +11,27 @@
 
 #include "HTTPRequest.hpp"
 #include "factoryRequestImplemetator.hpp"
+#include "json.hpp"
 #include "urlRequest.hpp"
+#include <algorithm>
+#include <string>
+#include <unordered_set>
 
 using wrapperType = cURLWrapper;
 
 void HTTPRequest::download(const URL& url,
                            const std::string& outputFile,
-                           std::function<void(const std::string&, const long)> onError)
+                           std::function<void(const std::string&, const long)> onError,
+                           const std::unordered_set<std::string>& httpHeaders)
 {
     try
     {
-        GetRequest::builder(FactoryRequestWrapper<wrapperType>::create())
-            .url(url.url())
-            .outputFile(outputFile)
-            .execute();
+        auto req {GetRequest::builder(FactoryRequestWrapper<wrapperType>::create())};
+
+        std::for_each(
+            httpHeaders.begin(), httpHeaders.end(), [&req](const std::string& header) { req.appendHeader(header); });
+
```

Comment:
we can add a function in `IRequestImplementator` and implement it in `cURLWrapper` to avoid this repeated code that is used in multiple functions
what do you think?
```suggestion
        req.appendHeaders(httpHeaders);

```

Reply:
Done. But not in those classes you've mentioned.